src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -143,10 +146,10 @@ public function validateDefinition($element, $objectDefinition = null, JsonPoint
      */
     protected function getProperty($element, $property, $fallback = null)
     {
-        if (is_array($element) /*$this->checkMode == self::CHECK_MODE_TYPE_CAST*/) {
-            return array_key_exists($property, $element) ? $element[$property] : $fallback;
-        } elseif (is_object($element)) {
-            return property_exists($element, $property) ? $element->$property : $fallback;
+        if (is_array($element) && array_key_exists($property, $element) /*$this->checkMode == self::CHECK_MODE_TYPE_CAST*/) {
```

Comment:
what do you think of `isset` instead of `array_key_exists` which is a lot faster.

this would mean you also need an additional call to `array_key_exists` in case it does allow caching of NULL values. In cases were most calls would already be catched by `isset($element[property])` this will speedup things a lot.

so the end result would render to something like

```php
if (is_array($element)) {
  if (isset($element[property]) || array_key_exists($property, $element)) {
    return $element[property];
  }
}
// ...
return $fallback;
```